src/argilla/client/workspaces.py

```diff
@@ -13,68 +12,303 @@
-        is provided, the current workspace will be used.
+    client: "httpx.Client"
+    id: UUID
+    name: str
+    users: Optional[List["WorkspaceUserModel"]] = None
+    inserted_at: datetime
+    updated_at: datetime
+
+    def __init__(
+        self, client: "httpx.Client", *, id: Union[str, UUID], name: str, inserted_at: datetime, updated_at: datetime
+    ) -> None:
+        """Initializes a `Workspace` instance. Ideally this class should not be instantiated
+        directly, instead you should use one of the following classmethods:
+            - `Workspace.create` to create a new workspace in Argilla
+            - `Workspace.from_name` to get an existing workspace from Argilla by its name
+            - `Workspace.from_id` to get an existing workspace from Argilla by its ID
 
         Args:
-            name: the name of the workspace to be managed. If not provided, the current
-                workspace will be used.
+            client: the `httpx.Client` initialized to interact with the Argilla API.
+            name: the name of the workspace to be managed.
             id: the ID of the workspace to be managed.
+            inserted_at: the datetime when the workspace was created.
+            updated_at: the datetime when the workspace was last updated.
+
+        Raises:
+            ValueError: if the workspace does not exist in the current Argilla account.
+            ValidationApiError: if the ID provided is not a valid UUID.
+            RuntimeError: if there was an error while retrieving the workspace/s from Argilla.
+
+        Examples:
+            >>> from argilla import rg
+            >>> workspace = rg.Workspace(
+            ...     client=rg.active_client().http_client.httpx,
+            ...     name="my-workspace",
+            ...     id="...",
+            ...     inserted_at=datetime.datetime(2021, 8, 31, 10, 0, 0),
+            ...     updated_at=datetime.datetime(2021, 8, 31, 10, 0, 0),
+            ... )
         """
+        self.client = client
+        self.id = id if isinstance(id, UUID) else UUID(id)
         self.name = name
-        self.id = id or None
+        self.inserted_at = inserted_at
+        self.updated_at = updated_at
+
+    @property
+    def users(self) -> List["WorkspaceUserModel"]:
+        """Returns the list of users linked to the workspace.
+
+        Returns:
+            A list of `WorkspaceUserModel` instances.
+        """
+        if not hasattr(self, "__users") or self.__users is None:
+            self.__users = workspaces_api.list_workspace_users(self.client, self.id).parsed
+        return self.__users
+
+    def __str__(self) -> str:
+        return (
+            f"Workspace(id={self.id}, name={self.name}, users={self.users}, insterted_at={self.inserted_at},"
+            f" updated_at={self.updated_at}"
+        )
+
+    # TODO(alvarobartt): also allow user addition via name (which is more user-friendly)
+    def add_user(self, id: str) -> None:
```

Comment:
we can explicit the id is for a user
```suggestion
    def add_user(self, user_id: str) -> None:
```